app/controllers/users_controller.rb

```diff
@@ -1,7 +1,21 @@
 class UsersController < ApplicationController
+  before_action :find_user, only: [:show]
+
   def index
+    @users = User.all
   end
 
   def show
+    if @user.nil?
+      flash[:status] = :failure
+      flash[:result_text] = "User not found!"
+      redirect_to users_path # AO: Where should we redirect this?
+    end
+  end
+
+  private
+
+  def find_user
```

Comment:
let's check to see if this should be in a separate application_controller.rb file?